src/Agent.Worker/Build/FileContainerServer.cs

```diff
@@ -196,79 +203,90 @@ private async Task<UploadResult> UploadAsync(IAsyncCommandContext context, int u
                 token.ThrowIfCancellationRequested();
                 try
                 {
-                    using (FileStream fs = File.Open(fileToUpload, FileMode.Open, FileAccess.Read, FileShare.Read))
```

Comment:
This diff is pretty bad. The only notable change here is removing the encompassing `using` and moving it to the `else` case for not uploading to blob.